packages/libs/src/services/solana/SolanaWeb3Manager.ts

```diff
@@ -702,23 +720,28 @@ export class SolanaWeb3Manager {
       data: Buffer.from(data)
     })
 
-    const instructions = [
-      transferInstruction,
-      paymentRouterInstruction,
-      memoInstruction
-    ]
-    return instructions
+    instructions.push(paymentRouterInstruction, memoInstruction)
+
+    const recentBlockhash = (await this.connection.getLatestBlockhash())
+      .blockhash
+    const transaction = new Transaction({
+      feePayer: this.feePayerKey,
+      recentBlockhash: recentBlockhash
+    }).add(...instructions)
+    if (!isSenderUserBank && senderKeypair) {
+      transaction.partialSign(senderKeypair)
+    }
+    return transaction
   }
 
-  async purchaseContentWithPaymentRouter({
+  async purchaseContentWithPaymentRouterFromSolanaRootAccount({
```

Comment:
This is starting to get a bit hairy... 😨 Things will get better when we do all purchases through payment router I guess...

Reply:
not necessarily bc we'd still need different logic for the user bank vs solana root account flows. but at least the names will get shorter.